src/main/groovy/com/bmuschko/gradle/docker/internal/RegistryAuthLocator.groovy

```diff
@@ -130,16 +139,112 @@ class RegistryAuthLocator {
 
         } catch(Exception ex) {
             logger.error('Failure when attempting to lookup auth config ' +
-                '(docker repository: {}, configFile: {}. ' +
+                '(docker repository: {}, configFile: {}). ' +
                 'Falling back to docker-java default behaviour',
                 repository,
                 configFile,
                 ex)
         }
-        defaultAuthConfig
+        return null
+    }
+
+    /**
+     * Gets all authorization information
+     * using $DOCKER_CONFIG/.docker/config.json file
+     * If missing, an empty AuthConfigurations object is returned
+     * @return AuthConfigurations object containing all authorization information,
+     * or an empty object if not available
+     */
+    AuthConfigurations lookupAllAuthConfigs() {
```

Comment:
Shouldn't we be able to reuse logic from `lookupAuthConfig`? Seems like we are duplicating logic.

Reply:
I might look into additional refactorings later.